source/streaming-mode/streaming-read.txt

```diff
@@ -344,12 +351,13 @@ The following example shows how to stream data from MongoDB to your console.
 
 .. important:: Inferring the Schema of a Change Stream
 
-   When the {+connector-short+} infers the schema of a DataFrame
-   read from a change stream, by default,
-   it uses the schema of the underlying collection rather than that
-   of the change stream. If you set the ``change.stream.publish.full.document.only``
-   option to ``true``, the connector uses the schema of the 
-   change stream instead.
+   If you set the ``change.stream.publish.full.document.only``
+   option to ``true``, the {+connector-short+} infers the schema of a ``DataFrame``
+   by using the schema of the scanned documents. If you set the option to
+   ``false``, you must specify a schema.
+   
+   Schema inference happens at the beginning of streaming, and does not take into
+   account collections that are created during streaming.
```

Comment:
The information from the "Performance considerations" section from "Documentation Changes Summary" is included neither to this note, not anywhere else in the PR.

Reply:
Added to [the section here](https://preview-mongodbjordansmith721.gatsbyjs.io/spark-connector/DOCSP-36546-multiple-collections-support/streaming-mode/streaming-read-config/#specifying-multiple-collections-in-the-collection-property)

Reply:
We should change the new text added in the PR:

> When streaming from multiple collections, the connector reads from each collection sequentially. Streaming from a large number of collections cause slower performance.

I should clarify the "Performance considerations" section from from "Documentation Changes Summary", I apologize for not expressing this information more verbousely and clearly on the first attempt.

The "Performance considerations" section is a subsection of the "Schema inference" section, i.e., this section talks about the performance consideration with regard to schema inference. Note also that the term "sampling" is used alongside the term "scanning" (a.k.a. "reading"): _"When scanning multiple collections, each collection is sampled sequentially."_ When the connector infers a schema, it [`$sample`s](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) collections, and then infers the schema based on the sample documents. When multiple collections are involved, they are sampled sequentially (`$sample` works only with a single collection). If the connector is configured to read from multiple collections and to infer the schema, sampling them all sequentially may take noticeable time, i.e., schema inference may take noticeable time. However, once the schema is ready, there is no more sequential sampling and, consequently, no performance implications caused by it.